src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/service/ReportService.kt

```diff
@@ -323,4 +325,35 @@ class ReportService(
       reportBasic to getQuestions().map { it.toDto() }
     }
   }
+
+  /**
+   * Replaces one prisoner number in all report-related entities with another.
+   * NB: free text fields are _not_ changed
+   */
+  @Transactional
+  fun replacePrisonerNumber(removedPrisonerNumber: String, prisonerNumber: String): List<ReportBasic> {
+    // TODO: maybe free text fields should be replaced too? especially if we end up generating report titles automatically
```

Comment:
Interesting, would the title not be auto generated ?

Reply:
that's the point – so far the plan seems to be to not give users a chance to enter a title and there was talk of auto-generating one based on type and prisoner involvement etc. *if* that's what happens, this method ought to replace references in the title too.

i think at this stage it's best to leave the TODO msg since we don't know what titles will be like…

Reply:
(Unless titles have names instead of prisoner numbers...but then again, TBD)